File: src/main/java/com/github/sorusclient/client/ui/framework/constraint/Copy.java
package com.github.sorusclient.client.ui.framework.constraint;

import com.github.sorusclient.client.ui.framework.Component;
import com.github.sorusclient.client.util.Color;

public class Copy implements Constraint {

    private final double multiplier;

    public Copy() {
        this(1);
    }

    public Copy(double multiplier) {
        this.multiplier = multiplier;
    }

    @Override
    public double getXValue(Component.Runtime componentRuntime) {
        return 0;
    }

    @Override
    public double getYValue(Component.Runtime componentRuntime) {
        return 0;
    }

    @Override
    public double getWidthValue(Component.Runtime componentRuntime) {
        return componentRuntime.getHeight() * this.multiplier;
    }

    @Override
    public double getHeightValue(Component.Runtime componentRuntime) {
        return componentRuntime.getWidth() * this.multiplier;
    }

    @Override
    public double getCornerRadiusValue(Component.Runtime componentRuntime) {
        return 0;
    }

    @Override
    public double getPaddingValue(Component.Runtime componentRuntime) {
        return 0;
    }

    @Override
    public Color getColorValue(Component.Runtime componentRuntime) {
        return null;
    }

    @Override
    public String getStringValue(Component.Runtime componentRuntime) {
        return null;
    }

}

File: src/main/kotlin/com/github/sorusclient/client/ui/framework/constraint/Copy.kt
package com.github.sorusclient.client.ui.framework.constraint

import com.github.sorusclient.client.ui.framework.Component
import com.github.sorusclient.client.util.Color

class Copy @JvmOverloads constructor(private val multiplier: Double = 1.0) : Constraint {
    override fun getXValue(componentRuntime: Component.Runtime): Double {
        return 0.0
    }

    override fun getYValue(componentRuntime: Component.Runtime): Double {
        return 0.0
    }

    override fun getWidthValue(componentRuntime: Component.Runtime): Double {
        return componentRuntime.height * multiplier
    }

    override fun getHeightValue(componentRuntime: Component.Runtime): Double {
        return componentRuntime.width * multiplier
    }

    override fun getCornerRadiusValue(componentRuntime: Component.Runtime): Double {
        return 0.0
    }

    override fun getPaddingValue(componentRuntime: Component.Runtime): Double {
        return 0.0
    }

    override fun getColorValue(componentRuntime: Component.Runtime): Color {
        return null!!
    }

    override fun getStringValue(componentRuntime: Component.Runtime): String {
        return null!!
    }
}